src/txn/txn_ckpt.c

```diff
@@ -767,10 +764,93 @@ __checkpoint_prepare(
 	 */
 	WT_ASSERT(session, session->ckpt_handle_next == 0);
 	WT_WITH_TABLE_READ_LOCK(session, ret = __checkpoint_apply_all(
-	    session, cfg, __wt_checkpoint_get_handles, NULL));
+	    session, cfg, __wt_checkpoint_get_handles));
 	return (ret);
 }
 
+/*
+ * __txn_checkpoint_can_skip --
+ *	Determine whether it's safe to skip taking a checkpoint. This function
+ *	also parses out some configuration options and hands them back to the
+ *	caller - make sure it does that parsing regardless of the result.
+ */
+static int
+__txn_checkpoint_can_skip(WT_SESSION_IMPL *session,
+    const char *cfg[], bool *fullp, bool *use_timestampp, bool *can_skipp)
+{
```

Comment:
Having 3 boolean pointers as args makes the interface here clunky, especially since there is only a single caller of this function. Most of that is because of the config parsing. The interesting thing is that I don't see the removal of the same parsing elsewhere, so it is additional/duplicate work instead of moving the work.